components/sidebar/sidebar_service.cc

```diff
@@ -128,9 +129,23 @@ SidebarItem::BuiltInItemType GetBuiltInItemTypeForLegacyURL(
   return SidebarItem::BuiltInItemType::kNone;
 }
 
-std::vector<SidebarItem> GetDefaultSidebarItems() {
+bool IsAllowedDefaultSidebarItemType(
+    const SidebarItem::BuiltInItemType& item_type,
+    PrefService* prefs) {
+  switch (item_type) {
+    case SidebarItem::BuiltInItemType::kWallet:
+      return brave_wallet::IsAllowed(prefs);
+    default:
+      return true;
+  }
+}
+
+std::vector<SidebarItem> GetDefaultSidebarItems(PrefService* prefs) {
```

Comment:
nit: how about making this as a member methods as it needs `prefs_`?

Reply:
Made `GetBuiltInItemForType` and `GetDefaultSidebarItems` member methods to avoid passing prefs around (in 905de1ebe624137ca3c425a6b26c5f7b4bb4fdd0)